lib/commands/misc/giverole.js

```diff
@@ -24,46 +24,84 @@ module.exports = class GiveRoleCommand extends Command {
 		this.queues = {};
 	};
 
+	grantRole(msg, role, channel) {
+		msg.member.addRole(role);
+		channel.send(`${msg.author} now has the ${role.name} role!`);
+	}
+
+	giveMemberRole(msg, role, channel, userRoles) {
+		// Only allows this to run if the user is not already a Member
+		if (!userRoles.has(role.id)) {
+			msg.member.addRole(role);
+
+			channel.send(`Welcome to the server, ${msg.author}! <:wutGiga:297897855727697921>`);
+		}
+	}
+
+	giveColorRole(msg, role, channel, userRoles) {
+		// Confirms that the user doesn't already have the color role
+		if (!userRoles.has(role.id)) {
```

Comment:
one final change! `giveColorRole` and `giveRegionRole` both checks that the user has the role or not, you could move this out to the `run` function, before line 99 where you `if (COLOR_ROLES.includes(roleArg)) {`

this way you only check it once instead of two times in total for each of those functions.